transport/src/main/java/io/netty/channel/local/LocalChannel.java

```diff
@@ -506,5 +456,48 @@ public void connect(final SocketAddress remoteAddress,
             LocalServerChannel serverChannel = (LocalServerChannel) boundChannel;
             peer = serverChannel.serve(LocalChannel.this);
         }
+
+        @Override
+        public void register0(LocalEventLoop eventLoop) {
+            // Check if both peer and parent are non-null because this channel was created by a LocalServerChannel.
+            // This is needed as a peer may not be null also if a LocalChannel was connected before and
+            // deregistered / registered later again.
+            //
+            // See https://github.com/netty/netty/issues/2400
+            if (peer != null && parent() != null) {
+                // Store the peer in a local variable as it may be set to null if doClose() is called.
+                // See https://github.com/netty/netty/issues/2144
+                final LocalChannel peer = LocalChannel.this.peer;
+                state = State.CONNECTED;
+
+                peer.remoteAddress = parent() == null ? null : parent().localAddress();
```

Comment:
unless parent can change after line 467, I think this conditional is not needed.

Reply:
The code was just moved... So I would prefer to not adjust it here and do a followup pr if needed